include/boost/utility/string_view.hpp

```diff
@@ -245,12 +245,21 @@ namespace boost {
               return npos;
             if (s.empty())
               return pos;
-            const_iterator iter = std::search(this->cbegin() + pos, this->cend(),
+            const charT* first_char_ptr = traits::find(ptr_ + pos, len_ - pos, s[0]);
```

Comment:
I'm not 100% happy with this one.  The reason is that `std::search` will use `operator=`, and `traits::find` will use "whatever is supplied by the user" - and they may not match.   Also, this "find the first match, then compare the rest" is what most implementations of `search` do.  On the other hand, should we even be using `std::search` here - or should we use the traits? I need to research this.

Reply:
Maybe we could call `std::search` and pass `traits::eq` as the comparison predicate.

Reply:
> Maybe we could call std::search and pass traits::eq as the comparison predicate.
This was already the case.

I'd like to avoid std::search because the traits stuff should be faster. I once measured `string.find(char)` against `std::find` and the first is like 10x faster.

One possible way is to find first matching character with `traits::find` and check the rest with `traits::compare`. if match, return, if no match repeat the search continuing immediately after `first_char_ptr`.
This way we completely avoid std::search and rely only on traits. `traits.find` is optimized to `memchr`, and traits.compare is probably optimized to `memcmp`.